src/scripting/lua_globals.cc

```diff
@@ -72,7 +72,44 @@ files name.
    instead uses our own ``format`` function. This allows for better control of the
    formatting as well as reordering of arguments which is needed for proper localisation.
 
+   Use ``bformat()`` whenever if your string contain values from variables.
+
    :returns: The formatted string.
+
+   ``bformat()`` is used with the global :func:`_(str) function <_>`. Examples:
+
+   .. code-block:: lua
+
+      local str    = "widelands"
+      local number = 5
+
+      _("This is a string: %1$s"):bformat(str)             -- $s = string
+      _("This is a number: %1$d"):bformat(number)          -- $d = number (integer)
+
+      -- %1 refers to the first,
+      -- %2 to the second placeholder and so on:
+
+      _("%1$s has %2$d regular tribes"):bformat(str, number)
+
+      -- Formatting numbers with two digits:
+      local first  = 2
+      local sec    = 10
+      local third  = 5
+
+      _("%1$02d:%2$02d - %3$02d"):bformat(first, sec, third) -- result: 02:10 - 05
+
+
+   Together with :doc:`richtext <autogen_auxiliary_richtext>`:
+
+   .. code-block:: lua
+
+      local winner, points = get_winner(wl.Game().players)
+
+      p(_("%1$s has won the game with %2$d points!"):bformat(winner, points))
+
+   If you need more options look at
+   `LUA string.format <https://www.lua.org/manual/5.1/manual.html#pdf-string.format>`_
+   and the `c++ printf function <https://cplusplus.com/reference/cstdio/printf/>`_ .
 */
```

Comment:
In my opinion this documentation is mainly for add-on authors. Other developers may know the full documentation. I thought to link to this file, but i decided against this, because if the file get renamed, the doc get moved, or the whole code get a new hoster, this link get dead.

The section ``bformat`` covers now 99,9% of all use cases for add-on authors, imho.